man/systemd.exec.xml

```diff
@@ -1699,6 +1699,25 @@ BindReadOnlyPaths=/var/lib/systemd</programlisting>
         logging.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>RestrictFileSystems=</varname></term>
+
+        <listitem><para>Restricts the set of filesystems processes of this unit can open files on. Takes a space-separated
+        list of filesystem names. Any filesystem listed is made accessible to the unit's processes, access to filesystem
+        types not listed is prohibited (allow-listing). If the empty string is assigned, access to filesystems is not
+        restricted. This option may appear more than once, in which case the namespace
+        types are merged by <constant>OR</constant>.</para>
+
+        <para>Example: if a unit has the following,
+        <programlisting>RestrictFileSystems=ext4 tmpfs
+RestrictFileSystems=ext2 ext4</programlisting>
+        then <constant>ext4</constant>, <constant>tmpfs</constant>, and <constant>ext2</constant> are set.</para>
+
+        <para>Note that this setting might not be supported on some systems (for example if the LSM eBPF hook is
+        not enabled in the underlying kernel or if not using the unified control group hierarchy). This setting
+        will fail in that case.</para></listitem>
```

Comment:
i figure sooner or later we need some kind of grouping here. similar to the allowlisting we have for syscalls. i.e. `@basic-api` for procfs + sysfs, then maybe `@common-block` for ext4, xfs, btrfs; `@network` for well-known network file systems, and so on

Reply:
(the `systemd-analyze filesystems` command i proposed elsewhere might be the right tool to show these groups)

Reply:
And of course  is_network_fs(),  is_temporary_fs(), fstype_is_network(), fstype_is_api_vfs() should be updated to use the same database...